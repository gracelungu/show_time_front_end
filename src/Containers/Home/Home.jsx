import React, { Component } from "react";
import "./Home.scss";

import Header from "../../Components/Header/Header";
import Genre from "../../Components/Genre/Genre";
import Classic from "../../Components/Classic/Classic";

class Home extends Component {
  render() {
    return (
      <section>
        <Header />

        <section className="body flex">
          <Genre />
          <div className="focus">
            <Classic />
          </div>
        </section>
      </section>
    );
  }
}

export default Home;
