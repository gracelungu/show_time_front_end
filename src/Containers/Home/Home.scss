@import "../../variables.scss";

.body {
  padding: 120px $sidePadding 10px;
  color: rgba(250, 250, 250, 0.4);
  overflow-x: hidden;
}

.body .focus {
  width: 60vw;
}

@media only screen and (max-width: 768px) {
  $sidePadding: 15px;
  .body {
    padding: 80px $sidePadding 10px;
    padding-left: 0;
    padding-right: 0;
  }
  .body .focus {
    width: 100%;
  }
}
