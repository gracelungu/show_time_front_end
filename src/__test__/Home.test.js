import React from "react";
import thunk from "redux-thunk";
import { mount } from "enzyme";
import { BrowserRouter as Router } from "react-router-dom";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import Home from "../Containers/Home/Home";

let wrapper, store;
const props = {};
const mockStore = configureMockStore([thunk]);
const storeStateMock = {
  index: { navToggle: true },
  auth: {
    user: {},
    loading: false,
    error: null,
    loggedIn: false
  },
  profile: {
    user: {},
    loading: false,
    error: null
  }
};

describe("Home.jsx", () => {
  beforeEach(() => {
    store = mockStore(storeStateMock);
    wrapper = mount(
      <Provider store={store}>
        <Router>
          <Home {...props} />
        </Router>
      </Provider>
    );
  });

  test("should render Home.jx", () => {
    expect(wrapper.find(".body").length).toBe(1);
  });
});
