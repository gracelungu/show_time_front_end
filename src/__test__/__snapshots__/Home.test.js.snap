// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.jsx should render Home.jx 1`] = `ShallowWrapper {}`;
